lib/inertia_rails/controller.rb

```diff
@@ -55,6 +65,53 @@ def _inertia_shared_data
           end.freeze
         end
       end
+
+      private
+
+      def push_to_inertia_share(**attrs, &block)
+        @inertia_share ||= []
+        @inertia_share << attrs.freeze unless attrs.empty?
+        @inertia_share << block if block
+      end
+
+      def extract_inertia_share_options(props)
+        options = props.slice(:if, :unless, :only, :except)
+
+        return options if options.empty?
+
+        if props.except(:if, :unless, :only, :except).any?
+          raise ArgumentError, "You must not mix shared data and [:if, :unless, :only, :except] options, pass data as a hash or a block."
+        end
+
+        extract_inertia_share_option(options, :only, :if)
```

Comment:
Could we add some indication that these methods are mutating their argument? maybe just a `!` suffix (even though ruby is super vague on what that means 😄 )

Reply:
That's a small thing, but I will stand my ground 😂

I believe generally a `!` method will have a non-bang counterpart. Another point is that Rails itself doesn't use bang for this method: https://github.com/rails/rails/blob/v7.2.0/actionpack/lib/abstract_controller/callbacks.rb#L98-L104

Reply:
Ok, I can cede on the `!` 😄 

Maybe there's a naming thing we could change here? Your example from rails is good, but I think maybe what throws me here is that we have `extra_inertia_share_options` above which is a pure function and then a very similarly named method here that mutates its arguments and has no return. Would something like `transform_inertia_share_option` make sense?

Reply:
Sounds good, updated 👍